topdown/lineage/lineage.go

```diff
@@ -2,22 +2,38 @@
 // Use of this source code is governed by an Apache2
 // license that can be found in the LICENSE file.
 
-package notes
+package lineage
 
 import (
 	"github.com/open-policy-agent/opa/topdown"
 )
 
-// Filter returns a filtered trace that contains Note events contained in the
-// trace. The filtered trace also includes Enter and Redo events to explain the
-// lineage of the Note events.
-func Filter(trace []*topdown.Event) (result []*topdown.Event) {
+// Notes returns a filtered trace that contains Note events and context to
+// understand where the Note was emitted.
+func Notes(trace []*topdown.Event) []*topdown.Event {
+	return Filter(trace, func(event *topdown.Event) bool {
+		return event.Op == topdown.NoteOp
+	})
+}
+
+// Fails returns a filtered trace that contains Fail events and context to
+// understand where the Fail occurred.
+func Fails(trace []*topdown.Event) []*topdown.Event {
+	return Filter(trace, func(event *topdown.Event) bool {
+		return event.Op == topdown.FailOp
+	})
+}
+
+// Filter returns a filtered copy of the trace. If the filter function returns
+// true, the event is kept. The filtered trace contains Enter and Redo events
+// for context.
+func Filter(trace []*topdown.Event, filter func(*topdown.Event) bool) (result []*topdown.Event) {
```

Comment:
Does `Filter` need to be public?

Reply:
Not really. I'm fine with unexporting it.